chore: update to 3.9.0 final

henryiii opened the pull request:
Python 3.9.0 was released yesterday. Nuget was fast, though Manylinux finally updated 5 hours ago ~~(do we pin a docker image? If not, this should just work?)~~. Bumping macOS and Windows to 3.9.0. The other requirements should possibly be bumped before a release is made with 3.9.0 final, I think (@joerick?).

Edit: Oops, they are pinned, just found that. Will update.

YannickJadoul commented:
@henryiii, there's a nice script to do that, `bin/update_dependencies.py`. Just comment out the `pip-compile` part, if you don't want to update the `constraints.txt` files.

henryiii commented:
That's what I'm working on running (though I assume updating the constraints is fine, latests updates might be helpful for 3.9?), but I'm running into`No such file or directory: './env27/bin/pip-compile'` and investigating.

YannickJadoul commented:
Those are virtualenvs that @joerick uses to update the package versions, I believe. I just comment out these lines when updating the images:

https://github.com/joerick/cibuildwheel/blob/630653b786790c3dbfbadb22eb2eaa9cd53d9e08/bin/update_dependencies.py#L13-L29

YannickJadoul commented:
Also, the PyPy failures should be fixed once #430 is merged.

henryiii commented:
I'll just comment out the constraints part for now, looks like I'd have to use pyenv and setup multiple Pythons to run that update; assuming one of you already has it (will do it eventually). Looks like the macOS tests are failing due to PyPy 7.3.1? Or am I misreading that?

henryiii commented:
By the way, do we need to have a 37 constraints file now? If you run it with 3.8, then `importlib-metadata==1.7.0` goes away.

YannickJadoul commented:
> assuming one of you already has it (will do it eventually).

I don't. I believe it's part of @joerick's steps when making a new release.

> Looks like the macOS tests are failing due to PyPy 7.3.1? Or am I misreading that?

Yup, related to #371, https://foss.heptapod.net/pypy/pypy/issues/3229 and https://foss.heptapod.net/pypy/pypy/merge_requests/726.

> By the way, do we need to have a 37 constraints file now? If you run it with 3.8, then `importlib-metadata==1.7.0` goes away. And `zipp`.

We probably do, then, yes :-(

henryiii commented:
@joerick Would you like me to add a env37/constraints37 setup, and start assuming that the "main" one runs 3.8 (which should be identical to 3.9, so a "38" constraints is still not needed)?

@YannickJadoul What do we do about PyPy? And why is it just now breaking?

YannickJadoul commented:
> env37/constraints37 setup

Maybe you can add this to the list of versions in `bin/update_dependencies.py`, such that @joerick will get a (not-so-gentle) reminder, next time these dependencies are updated?

> @YannickJadoul What do we do about PyPy? And why is it just now breaking?

That I don't really know. The issue was linked to versions of the libraries installed in the environment. Did Azure update their images? (Always blame CI updates, when possible ;-) )

YannickJadoul reviewed (approved):
After merging #430, things should turn green. At any rate, the failures are not related to this PR.

henryiii commented:
@joerick be sure to check 501ac92, I've added a 3.7 constraints file there and updated the docs on setting up a environment to run the update step (and, obviously, I ran the update step).

YannickJadoul reviewed (approved):
Hurray! :-)

joerick commented:
Niiiiice work @henryiii ! Thanks for doing that, and updating the docs :) Yep, this looks good to me, I'll try to push a release soon so we have a Python 3.9.0 version out there.

henryiii commented:
Any idea why one AppVeyor job basically timed out? https://ci.appveyor.com/project/joerick/cibuildwheel/builds/35599358/job/hiq0jff3cca76c4f 

Can someone restart it?

mayeut commented on cibuildwheel/resources/pinned_docker_images.cfg:
```diff
@@ -1,22 +1,22 @@
 
 [aarch64]
-manylinux2014 = quay.io/pypa/manylinux2014_aarch64:2020-09-26-7f7a14f
+manylinux2014 = quay.io/pypa/manylinux2014_aarch64:2020-10-05-2f3b046
```
The aarch64 build failed on travis-ci fi-or the manylinux2014 aarch64 image. I restarted it and it should now be available.
```suggestion
manylinux2014 = quay.io/pypa/manylinux2014_aarch64:2020-10-06-b2ca7a1
```

henryiii replied:
Good catch, didn't notice that! Reran `./bin/update_dependencies.py` and indeed, this is the only change.

henryiii commented:
Passed and several approvals, so merged! :)
